control/cli.py

```diff
@@ -293,10 +294,10 @@ def create_listener(self, args):
 
     @cli.cmd([
         argument("-n", "--subnqn", help="Subsystem NQN", required=True),
-        argument("-g", "--gateway-name", help="Gateway name", default=""),
+        argument("-g", "--gateway-name", help="Gateway name", default=socket.gethostname()),
```

Comment:
I am not sure I like this default. Note, you may have cli on a different host than the gateway. And actually it may be confusing. I would prefer one of the options below:

1) make `--gateway-name` mandatory
2) still have it optional, but if it is not specified use the name from ceph-nvme.conf (and fail if it is not specified there)
3) don't change anything on the cli side, but on the the gateway side use that gateway name if the gateway param is empty in the request (I like this less as it still may be confusing)

Reply:
I did not consider the situation of cli on different hosts. If on the same host, I think Option 3 would be simpler as it does not require changing existing commands. But if we consider that cli is on different hosts, then Option 1 may be a better choice and more in line with previous comments https://github.com/ceph/ceph-nvmeof/issues/87#issuecomment-1533870960, but it is more stringent and will change existing commands. So, I revised the `README`. Moreover, gatewatt defaults to using the host name as the gateway name, which may also be confusing.  https://github.com/ceph/ceph-nvmeof/blob/e783513fc851d46254e8b34466552a1069046260/control/grpc.py#L44

I made changes according to Scheme 1. The server does not need to make additional judgments because I removed the default value assignment, even if some one talk to the gateway directly, not using cli. spdk will report the error.

Reply:
Maybe Scheme 2 can be implement like this? keep `default=""`, then handle in function.
```
    @cli.cmd([
        argument("-g", "--gateway-name", help="Gateway name", default=""),
    ])
    def create_listener(self, args):
        if not args.gateway_name:
            gateway_name = self.gateway_name
            if not gateway_name:
                raise Exception("gateway.addr option is not set")
        try:
            req = pb2.create_listener_req(
                gateway_name=gateway_name,
            )
```

Reply:
I actually don't quite understand one thing. Even if the cli and gateway are not on the same host, the subsystem can only be created on the gateway host. That must be using the gateway's gateway_name and IP. Why can't we simply use the gateway's default gateway_name and IP and store them in omap. Compared to the original code, there are only additional steps for storing IP, like Scheme 3. This may return to the initial discussion.

Reply:
I suppose you are talking about what I proposed as the option (3)? It may be confusing. Consider a case when a CLI (or some other management application like ceph dashboard) may randomly connect to any of the gateways to create a listener. Or just the fact that result will depend on what host you are running the command (to what gateway are connecting) may be confusing.

I think if a user knows the IP for the listener to create she will also know the gateway, so just requiring to specify both looks like the best option here.

Reply:
Clear. Thank you for your explanation. Please review again, it is now mandatory to input gateway_name and ip.